pkg/bindings/connection.go

```diff
@@ -298,11 +316,31 @@ func tcpClient(_url *url.URL) (Connection, error) {
 			}
 		}
 	}
+	transport := http.Transport{
+		DialContext:        dialContext,
+		DisableCompression: true,
+	}
+	if len(tlsCAFile) != 0 || len(tlsCertFile) != 0 || len(tlsKeyFile) != 0 {
+		logrus.Debugf("using TLS cert=%s key=%s ca=%s", tlsCertFile, tlsKeyFile, tlsCAFile)
+		transport.TLSClientConfig = &tls.Config{}
+		connection.tls = true
+	}
+	if len(tlsCAFile) != 0 {
+		pool, err := util.ReadCertBundle(tlsCAFile)
+		if err != nil {
+			return connection, fmt.Errorf("unable to read CA bundle: %w", err)
+		}
+		transport.TLSClientConfig.RootCAs = pool
+	}
+	if len(tlsCertFile) != 0 && len(tlsKeyFile) != 0 {
```

Comment:
The condition here do not seem to line up perfectly.

If only `tlsCertFile` is set then the first condition matches and says tls is used but then we never add the the key on the last one. That case should return a hard error I would assume.

Reply:
My intent was to have that check performed in the `cmd/podman/...` packages, but on reflection it makes sense to also perform the same check here, as that wouldn't account for the env vars. Done.